qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/QuteProjectRegistry.java

```diff
@@ -229,48 +239,109 @@ private CompletableFuture<ResolvedJavaTypeInfo> resolveJavaType(String javaTypeN
 			// It's a primitive type like boolean, double, float, etc
 			return primitiveType;
 		}
-		if (StringUtils.isEmpty(javaTypeName) || StringUtils.isEmpty(projectUri)) {
-			return RESOLVED_JAVA_CLASSINFO_NULL_FUTURE;
+
+		String projectUri = project.getUri();

```

Comment:
I think it might be a good idea to split this method up into multiple methods to make it more legible

Reply:
Is it better?